test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -279,23 +281,43 @@ public void emailSignInThrowsConsentRequired() {
         study.setIdentifier(STUDY_ID);
         doReturn(TOKEN).when(cacheProvider).getString(CACHE_KEY);
         doReturn(study).when(studyService).getStudy(STUDY_ID);
-        doReturn(account).when(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        doReturn(account).when(accountDao).getAccountAfterAuthentication(study, RECIPIENT_EMAIL);
         
         service.emailSignIn(CONTEXT, SIGN_IN);
     }
     
     @Test
-    public void emailSignInChangesPassword() {
-        doReturn(TOKEN).when(cacheProvider).getString(CACHE_KEY);
-        doReturn(account).when(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
-        doReturn(PARTICIPANT).when(participantService).getParticipant(study, account, false);
+    public void reauthentication() {
+        ((GenericAccount)account).setId(USER_ID);
+
+        StudyParticipant participant = new StudyParticipant.Builder().withEmail(RECIPIENT_EMAIL).build();
         doReturn(CONSENTED_STATUS_MAP).when(consentService).getConsentStatuses(any());
+        doReturn(account).when(accountDao).reauthenticate(study, REAUTH_REQUEST);
+        doReturn(participant).when(participantService).getParticipant(study, account, false);
+        service.reauthenticate(study, CONTEXT, REAUTH_REQUEST);
```

Comment:
Verify that returned session contains reauthToken?

Do we also need this test for signIn and emailSignIn?

Reply:
Added this for all three paths.